### PR TITLE
Bugfix/ipv6 handling

### PR DESCRIPTION
**Pull request by metelik**
Changes:
- Fixing IPv6 parse function - IPv6 address is a tuple of 8 hextets (not 6)
- Adding AAAA class host-name resolution (:inet6)
- Adding switch between :inet and :inet6 address families for UDP client's socket opening based on the server's address

### CODE REVIEW

**Review by vkatsuba — COMMENTED** [review]
Hi. Thanks for your contribution! Please run GitHub Actions manually in your fork and provide here the link to success CI.
How to:

Open https://github.com/se-apc/eradius
Select tab Actions - https://github.com/se-apc/eradius/actions
Select tab CI
Find button Run workflow
Select in Run workflow branch master and click Run workflow

**Review by vkatsuba — COMMENTED** [review]
Please keep one code style - use 4 spaces instead of 2.

**Review by vkatsuba — COMMENTED** [review]
If you can, please provide test cases.

File: src/eradius_client.erl [hunk 1]
```diff
@@ -62,19 +62,29 @@ send_request(NAS, Request) ->
 %   If no answer is received within the specified timeout, the request will be sent again.
 -spec send_request(nas_address(), #radius_request{}, options()) ->
     {ok, binary(), eradius_lib:authenticator()} | {error, 'timeout' | 'socket_down'}.
-send_request({Host, Port, Secret}, Request, Options) 
+send_request({Host, Port, Secret}, Request, Options)
```
Comment (RoadRunnr) [inline src/eradius_client.erl:65]: please, don't mix pure formatting and white space changes with functional changes.

If you think the whitespace problems need fixing, create a separate pull request for that.

**Review by RoadRunnr — CHANGES_REQUESTED** [review]
You PR contains a merge commit. Please make sure that you only include you changes in the PR  and that the history is linear.

**Comment by metelik** [comment]
> 
> 
> If you can, please provide test cases.

Sure thing. I shall need to revisit some tests that have already been present due to the APIs changes. The functionality itself I had tested manually. I am not an Erlang major (not main main programming language)  so perhaps a joint effort in the expansion of the available test suite would be well appreciated.

**Comment by metelik** [comment]
> 
> 
> You PR contains a merge commit. Please make sure that you only include your changes in the PR and that the history is linear.

QQ on this one. I shall have been probably merging the current master branch into our fork in any way so another merge commit shall appear. Would the commits' squashing into one do? I am asking because such a synchronisation with the upstream master can be advantageous on many levels. OOC what would be the reasons behind your disliking that?

**Comment by vkatsuba** [comment]
Moved to https://github.com/travelping/eradius/pull/234

**Pull request closed without merging**